src/ardrive.ts

```diff
@@ -80,28 +126,39 @@ export class ArDrive extends ArDriveAnonymous {
 		password: string,
 		destinationFileName?: string
 	): Promise<ArFSResult> {
+		const winstonPrice = await this.walletDao.getWinstonPriceForBytes(this.getFileSize(filePath));
+		const communityWinstonTip = await this.communityOracle.getCommunityWinstonTip(winstonPrice);
+		const totalWinstonPrice = (+winstonPrice + +communityWinstonTip).toString();
+
+		if (!this.walletDao.confirmBalance(this.wallet, totalWinstonPrice)) {
+			throw new Error('Not enough AR for file upload..');
+		}
+
+		// TODO: Add interactive confirmation of AR price estimation
```

Comment:
When?

Reply:
Its a phase 1.5 thing once we have an interactivity pattern established. The current CLI will ask the user (can be set auto-accept) to confirm a price estimation before sending the transactions.